src/main/java/Duke.java

```diff
@@ -1,10 +1,229 @@
+            num++;
+            System.out.println("Now you have " + num + " tasks in the list.");
+        } catch (StringIndexOutOfBoundsException e) {
+            System.out.println("\u2639 The description of a event cannot be empty.");
+        } catch (DukeException e) {
+            System.out.println("\u2639 The date/time of a event cannot be empty.");
+        }
+        return num;
+    }
+
+
+    //Delete the task
+    private static int deleteTask(ArrayList<Task> list, String command, int num) {
+        try {
+            String[] word = command.split(" ");
+            int index = Integer.parseInt(word[1]) - 1;
+            if (index >= num) {
+                throw new ArrayIndexOutOfBoundsException();
+            }
+            System.out.println("Noted. I've removed this task: ");
+            System.out.println(" " + list.get(index));
+            list.remove(index);
+            num--;
+            System.out.println("Now you have " + num + " tasks in the list.");
+        } catch (ArrayIndexOutOfBoundsException e) {
+            System.out.println("\u2639 The number is not in the list");
+        }
+        return num;
+    }
+
+    private static void writeToFile(String filePath, ArrayList<Task> list) {
+        try {
+            FileWriter fw = new FileWriter(filePath);
+            for (Task t : list) {
+                switch (t.getCategory()) {
+                case "T":
+                    fw.write(t.getCategory() + "|" + t.getIsDone() + "|" + t.getDescription() + System.lineSeparator());
+                    break;
+                case "D":
+                    fw.write(t.getCategory() + "|" + t.getIsDone() + "|" + t.getDescription() + "|" + t.getBy()+ System.lineSeparator());
+                    break;
+                case "E":
+                    fw.write(t.getCategory() + "|" + t.getIsDone() + "|" + t.getDescription() + "|" + t.getAt()+ System.lineSeparator());
+                    break;
+                default:
+                    break;
+                }
+            }
+            fw.close();
+        } catch (IOException e) {
+            System.out.println("Something went wrong: " + e.getMessage());
+        }
+    }
+
+    private static int importFileContents(String filePath, ArrayList<Task> list) {
+        int num = 0;
+        try {
+            File d = new File("data");
+            if (d.mkdir()) {
+                System.out.println("Directory created successfully");
+            }
+            File f = new File(filePath); // create a File for the given file path
+            Scanner s = new Scanner(f); // create a Scanner using the File as the source
+            if (f.createNewFile()) {
+                throw new IOException();
+            }
+            while (s.hasNext()) {
+                String[] words = s.nextLine().split("\\|");
+                switch (words[0]) {
+                case "T":
+                    list.add(new Todo(words[2]));
```

Comment:
I think that it is best to avoid magic literals. In other words, create a constant for "0", "1", "2" etc. This will make code readability better